sycl/test/on-device/spec_const/multiple-usages-of-composite.cpp

```diff
@@ -0,0 +1,76 @@
+// UNSUPPORTED: cuda
+//
+// RUN: %clangxx -fsycl %s -o %t.out
+// RUN: %RUN_ON_HOST %t.out | FileCheck %s
+// RUN: %CPU_RUN_PLACEHOLDER %t.out %CPU_CHECK_PLACEHOLDER
+// RUN: %GPU_RUN_PLACEHOLDER %t.out %GPU_CHECK_PLACEHOLDER
+//
+// The test checks that multiple usages of the same specialization constant
+// works correctly with: toolchain processes them correctly and runtime can
+// correctly execute the program.
+//
+// CHECK: --------> 1
+
+#include <CL/sycl.hpp>
+
+using namespace cl::sycl;
+
+class sc_kernel_t;
+
+namespace test {
+
+struct pod_t {
+  float x;
+  float y;
+};
+
+template <typename T> class kernel_t {
+public:
+  using sc_t = sycl::ONEAPI::experimental::spec_constant<pod_t, sc_kernel_t>;
+
+  kernel_t(const sc_t &sc, cl::sycl::stream &strm) : sc_(sc), strm_(strm) {}
+
+  void operator()(cl::sycl::id<1> i) const {
+    strm_ << "--------> " << sc_.get().x << sycl::endl;
+  }
+
+  sc_t sc_;
+  cl::sycl::stream strm_;
+};
+
+template <typename T> class kernel_driver_t {
+public:
+  void execute(const pod_t &pod) {
+    device dev = sycl::device(default_selector{});
+    context ctx = context(dev);
+    queue q(dev);
+
+    cl::sycl::program p(q.get_context());
+    auto sc = p.set_spec_constant<sc_kernel_t>(pod);
+    p.build_with_kernel_type<kernel_t<T>>();
+
+    q.submit([&](cl::sycl::handler &cgh) {
+      cl::sycl::stream strm(1024, 256, cgh);
+      kernel_t<T> func(sc, strm);
+
+      auto sycl_kernel = p.get_kernel<kernel_t<T>>();
+      cgh.parallel_for(sycl_kernel, cl::sycl::range<1>(1), func);
+    });
+    q.wait();
+  }
+};
+
+template class kernel_driver_t<float>;
+
+// Uncomment this to trigger the bug.
```

Comment:
Should the line 66 be commented? Or should the line 65 be removed?

Reply:
66 shouldn't be commented or the bug won't be triggered. But I agree that the comment can be improved - I posted a suggestion, please take a look

Reply:
Thanks, looks good to me.

Reply:
Fixed in e09b1b859bd88f8dc46b0aaad8e4bc21a7993b27